src/main/java/org/jala/efeeder/image/ImageCommand.java

```diff
@@ -38,8 +38,16 @@ public Out execute(In parameters) throws Exception {
         } else {
             path = parameters.getParameter("image_path") + "/" + imageName;
         }
-        
-        return readImageAndConvertToBytes(path, "jpg", "image/jpeg");
+        String extension = FilenameUtils.getExtension(path);
+		switch(extension){
+			case "gif":
+				return readImageAndConvertToBytes(path, "gif", "image/gif");
+			case "png":
+				return readImageAndConvertToBytes(path, "png", "image/png");
+			default:
+				System.err.println("entro al jpg");
+				return readImageAndConvertToBytes(path, "jpg", "image/jpeg");
+		}
```

Comment:
Maybe you can do try { return readImageAndConvertToBytes(path, extension, "image/[extension]"); } catch {System.err.println("entro al jpg");}